### ISSUE TITLE
Open images from app folders instead of www/?

### ISSUE THREAD

**Issue by mirko77** [keensoft/FullScreenImage-Cordova-Plugin#19]
I have a question, with some changes would it be possible to open images in other paths other than `www`? Are there any permission issues on Android or iOS to void it?

I am looking to open images stored in the cache directory of both platforms:

`cordova.file.cacheDirectory` - Directory for cached data files or any files that your app can re-create easily. The OS may delete these files when the device runs low on storage, nevertheless, apps should not rely on the OS to delete files in here. (iOS, Android, BlackBerry 10)

Thanks!

**Comment by slaptot** [comment]
Hi @mirko77,
it´s a good enhancement and could be good in order to avoid store permanent files. Please try it and let me know

**Comment by shatle** [comment]
Can i open cache images now? Thanks.

**Comment by kschio** [comment]
Would be great to be able to open images from the cache/tmp folder.

**Comment by danicastanos** [comment]
+1

**Comment by scanniza** [comment]
Or just from any source! I have remotely hosted images I want to show!

**Comment by TheBosZ** [comment]
I just submitted a PR that adds this feature.

**Comment by michelecos** [comment]
For me it works on Android, with images from data folders of the application, but it cannot go outside www on iOS.

**Comment by arneson** [comment]
Yeah, not sure if this is fixed, but I ended up doing a workaround converting the image to base64:

```
convertImgToBase64(url,function(dataUrl){
    window.FullScreenImage.showImageBase64(dataUrl, ' ', "jpg");
});

function convertImgToBase64(url, callback, outputFormat){
    var img = new Image();
    img.crossOrigin = 'Anonymous';
    img.onload = function(){
        var canvas = document.createElement('CANVAS');
        var ctx = canvas.getContext('2d');
        canvas.height = this.height;
        canvas.width = this.width;
        ctx.drawImage(this,0,0);
        var dataURL = canvas.toDataURL();
        callback(dataURL.substring(dataURL.indexOf(',')+1));
        canvas = null; 
    };
    img.src = url;
  }
```

**Comment by michelecos** [comment]
Thank you very much. I was reasoning along the same lines.

Michele Costabile
michelecos@yahoo.com

**Comment by gaurav-chandra** [comment]
+1
